fix(es/codegen): Fix placing of comments of yield arguments

lewisl9029 opened the pull request:
<!-- Note: CI script will automatically rebase your PR so please do not rebase unless required -->

<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->

**Description:**

Looks like the bug I ran into had nothing to do with the changes in #7856, since it's reproducible without it. Looks like it might have only surfaced now because https://github.com/swc-project/swc/pull/7853 changed the default value of `jsc.minify.format.comments`? Added a minimal test case here with the expected result.

Here's the actual output:

```js
 export var padding = '';
 export function exec2({ commands }) {
     return __awaiter(this, void 0, void 0, function*() {
         for(let i2 = 0; i2 < commands.length; i2++){
             let command = commands[i2];
             yield // some-comment
             function({ command }) {
                 command();
             }({
                 command,
                 handleError
             });
         }
     });
 }
```

The comment ends up getting added after the yield, which makes the output invalid.

Going to see if I can figure out a fix tomorrow, but let me know if you have any ideas on where to start looking in the meantime!

kdy1 commented:
We may need `need_paren` like in https://github.com/swc-project/swc/blob/5ae2e810d8d8a2d23c9e81eae128fb72c5c1f09c/crates/swc_ecma_codegen/src/lib.rs#L2936-L2940 for `emit_yield_expr` too.

https://github.com/swc-project/swc/blob/5ae2e810d8d8a2d23c9e81eae128fb72c5c1f09c/crates/swc_ecma_codegen/src/lib.rs#L1882-L1886

lewisl9029 commented:
@kdy1 thank you so much for the suggestion, it worked great!
Added the fix and updated the expected output accordingly in https://github.com/swc-project/swc/pull/7858/commits/7d1bc9f0f56c02c57d92bbfeed45a4c3114b5a49

kdy1 reviewed (approved):
Thank you!

---

swc-bump:
 - swc_ecma_codegen
